XMonad/Hooks/DynamicLog.hs

```diff
@@ -585,6 +589,101 @@ pprWindowSetXinerama ws = "[" ++ unwords onscreen ++ "] " ++ unwords offscreen
         offscreen = map S.tag . filter (isJust . S.stack)
                         . sortOn S.tag $ S.hidden ws
 
+-- | Custom datatype for custom icons based on the state of the 'Workspace'
+-- For example,
+--
+-- > Icon "<bold>discord</bold>" "<bold>discord</bold" "discord" ""
+--
+-- Then you can add it to your IconSet
+--
+-- > icons :: IconSet
+-- > icons = mconcat
+-- >    [ className =? "discord" --> pure [Icon "<bold>discord</bold>" "<bold>discord</bold" "discord" ""]
+-- >    ]
+data Icon = Icon
+    { iconCurrent :: !String -- ^ If the 'Workspace' is the current workspace
+    , iconVisible :: !String -- ^ If the 'Workspace' is visible (Xinerama only)
+    , iconHidden :: !String -- ^ If the 'Workspace' isnt visible but still has windows
+    , iconHiddenNoWindows :: !String -- ^ If the 'Workspace' isnt visible and has no windows
+    }
+
+-- | The set of Icons to use
+type IconSet = Query [Icon]
+baseIconSet :: String -> Icon
+baseIconSet x =
+    Icon { iconCurrent = x
+         , iconVisible = x
+         , iconHidden = x
+         , iconHiddenNoWindows = x
+         }
+
+-- | Create an 'IconSet' from a 'String'
+appIcon :: String -> IconSet
+appIcon x = pure [baseIconSet x]
+
+-- | Dynamically shows 'WorkspaceId' based on the 'Window's inside the Workspace
+-- which 'Icon's to use are defined via an 'IconSet'
+-- to create an 'IconSet' make a query that returns a '[Icon]'.
```

Comment:
This will be rendered as a single sentence in the docs, but my brain (which has only spoken English for 25 years so I can't claim to be any expert) fails to parse it as such. It parses fine as three sentences, one per line. Is that what's intended? Can we add some periods and capital letters please?

Reply:
Yes sorry. Not your fault. I've fixed it